screeps-game-api/src/positions.rs

```diff
@@ -396,16 +517,33 @@ mod room_pos_serde {
         where
             D: Deserializer<'de>,
         {
-            let data = SerializedLocalRoomPosition::deserialize(deserializer)?;
-
-            Ok(LocalRoomPosition {
-                room_name: LocalRoomName {
-                    x_coord: data.room_x,
-                    y_coord: data.room_y,
-                },
-                x: data.x,
-                y: data.y,
-            })
+            if deserializer.is_human_readable() {
+                // an older version of the library always serialized as 'EfficientFormat' - this
+                // keeps compatibility with that.
```

Comment:
Do you really want backward compatibility at the cost of lines of code to be maintained?

Reply:
This is a good question!

I'm partial to backwards compatibility mostly because it's a pretty silent and easy to ignore change, and affects anyone storing LocalRoomPositions via `serde`?

We could definitely not do this, though, too.

Reply:
I haven't started using it yet, so choose whatever you think is best.
I think that striving for backward compatibility in a pre-1.0 release is just asking for dead code, but I can see the reasoning.

Reply:
Maybe we could make an issue for removing it next breaking release after this one?

I'm all in favor of breaking things, it's just nice to make them as visible as possible. I mean changing the serialization format is already technically a breaking change, it's just _slightly less breaking_ to accept the old one too on deserialization? ...